### ISSUE TITLE
DynamicSoundEffectInstance differences between XNA and FNA

### ISSUE THREAD

**Issue by prime31** [FNA-XNA/FNA#44]
So, I was just messing around with DynamicSoundEffectInstance and I grabbed [this old sample code](http://www.david-gouveia.com/portfolio/creating-a-basic-synth-in-xna-part-iii/) which is a very basic synthesizer. The code works perfectly with XNA 4 but with FNA it doesn't produce any audio.

Digging a bit deeper it appears that there is something different about the FNA `DynamicSoundEffectInstance.PendingBufferCount`. This particular section of code from the Synth class doesnt ever make it through the while loop except the first 2 times it is called.

```cs
            while (_instance.PendingBufferCount < 2)
            {
                SubmitBuffer();
            }
```

Removing the while loop and just submitting buffers every frame gets the audio playing but with a pretty massive delay on the order of 1 - 5 seconds depending on the buffer size.

**Comment by flibitijibibo** [comment]
We increment PendingBufferCount on SubmitBuffer, so something else might be happening. Trace that property's value in XNA and FNA and you should be able to find where we diverge. For delay, the game probably could just use a callback instead... Any idea why it uses this method instead?

**Comment by prime31** [comment]
I have no idea why it does things the way it does. I haven't delved too deep into the code yet. I wired up the BufferNeeded event and it doesnt fire as is.

Adding to this as I delve deeper. The first difference I am seeing is when dumping `PendingBufferCount` on XNA it is always 1 or 2. With FNA it is forever 2. If the while check is removed it just increments every frame.

OK. Found the anomaly but I don't know why just yet. The DynamicSoundEffectInstance `Update` method is never called. If I call it manually everything works perfectly with no delay and the BufferNeeded event fires.
